OpenRA.Mods.TS/Traits/SupportPowers/AttackOrderPower.cs

```diff
@@ -0,0 +1,111 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.Traits
+{
+	class AttackOrderPowerInfo : SupportPowerInfo
+	{
+		public override object Create(ActorInitializer init) { return new AttackOrderPower(init.Self, this); }
+	}
+
+	class AttackOrderPower : SupportPower, INotifyAddedToWorld, INotifyAttack
+	{
+		AttackOrderPowerInfo info;
+		AttackBase attack;
+
+		public AttackOrderPower(Actor self, AttackOrderPowerInfo info)
+			: base(self, info)
+		{
+			this.info = info;
+		}
+
+		public override void SelectTarget(Actor self, string order, SupportPowerManager manager)
+		{
+			Game.Sound.PlayToPlayer(manager.Self.Owner, Info.SelectTargetSound);
+			self.World.OrderGenerator = new SelectAttackPowerTarget(order, manager, info.Cursor, MouseButton.Left, attack);
+		}
+
+		public override void Charged(Actor self, string key)
+		{
+		}
+
+		public override void Activate(Actor self, Order order, SupportPowerManager manager)
+		{
+			base.Activate(self, order, manager);
+			attack.AttackTarget(Target.FromCell(self.World, order.TargetLocation), false, false, true);
+		}
+
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
+		{
+			attack = self.Trait<AttackBase>();
+		}
+
+		void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barrel)
+		{
+			self.World.IssueOrder(new Order("Stop", self, false));
```

Comment:
Ah, I can see you shifted the Delay to the SW, but `Burst should be supported here IMO.